### PR TITLE
fix(provider/openstack): Add nullcheck for caching autoscalling type 

### PR DESCRIPTION
**Pull request by emjburns**
Adding null check for autoscaling type to fix NPE while caching.

### CODE REVIEW

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy [hunk 1]
```diff
@@ -280,9 +273,9 @@ class OpenstackServerGroupCachingAgent extends AbstractOpenstackCachingAgent imp
       result.put('minSize', parameters.minSize ?: 0)
       result.put('maxSize', parameters.maxSize ?: 0)
       result.put('desiredSize', parameters.desiredSize ?: 0)
-      result.put('autoscalingType', parameters.autoscalingType.jsonValue())
+      result.put('autoscalingType', parameters.autoscalingType ? parameters.autoscalingType.jsonValue() : "")
       [up:parameters.scaleup, down:parameters.scaledown].each {
-        result.put("scale${it.key}".toString(), objectMapper.convertValue(it.value, ATTRIBUTES))
+          result.put("scale${it.key}".toString(), objectMapper.convertValue(it.value, ATTRIBUTES))
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy]: Was the additional indentation intentional here?

**Comment by duftler** [comment]
Needs a rebase.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy (hunk 1 above)
Comment (emjburns) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy]: Nope - removing and rebasing. Thanks!

**Comment by emjburns** [comment]
@duftler I've rebased, can you merge? Thanks!

**Comment by duftler** [comment]
Sorry I missed the @ mention.

**Comment by jrsquared** [comment]
Fixes https://github.com/spinnaker/spinnaker/issues/1558

**Pull request merged**